org.lflang/src/org/lflang/generator/JavaGeneratorUtils.java

```diff
@@ -32,6 +305,45 @@ public static void writeSourceCodeToFile(CharSequence code, String path) throws
         }
     }
 
+    /** If the mode is INTEGRATED (the code generator is running in an
+     *  an Eclipse IDE), then refresh the project. This will ensure that
+     *  any generated files become visible in the project.
+     */
+    public static void refreshProject(Mode compilerMode, String code) {
+        if (compilerMode == Mode.EPOCH) {
+            // Find name of current project
+            String id = "((:?[a-z]|[A-Z]|_\\w)*)";
+            Pattern pattern;
+            if (File.separator.equals("/")) { // Linux/Mac file separator
+                pattern = Pattern.compile(
+                    "platform:" + File.separator + "resource" + File.separator + id + File.separator);
+            } else { // Windows file separator
+                pattern = Pattern.compile(
+                    "platform:" + File.separator + File.separator + "resource" + File.separator + File.separator +
+                        id + File.separator + File.separator);
+            }
+            // FIXME: If we have to hunt through generated code to find this information, then maybe that's a sign
+            //  that our left hand isn't talking to our right.
+            Matcher matcher = pattern.matcher(code);
+            String projName = "";
+            if (matcher.find()) {
+                projName = matcher.group(1);
+            }
+            try {
+                IResource[] members = ResourcesPlugin.getWorkspace().getRoot().members();
+                for (IResource member : members) {
+                    // Refresh current project, or simply entire workspace if project name was not found
+                    if (projName.isEmpty() || projName.equals(member.getFullPath().toString().substring(1))) {
+                        member.refreshLocal(IResource.DEPTH_INFINITE, null);
+                        System.out.println("Refreshed " + member.getFullPath());
+                    }
+                }
+            } catch (IllegalStateException | CoreException e) {
+                System.err.println("Unable to refresh workspace: " + e);
+            }
+        }
```

Comment:
Extracting some location pattern from the generated code seems quite complicated.
If it is sufficient to update only the project the source LF file is located in, assuming that the generated files will be in the same project, it can be done via the Resource that is available in the generator.
~~~
URI uri = resource.getURI();
if (uri.isPlatformResource()) { // This condition should normally be met when running Epoch
    IResource member = ResourcesPlugin.getWorkspace().getRoot().findMember(uri.toPlatformString(true));
    member.getProject().refreshLocal(IResource.DEPTH_INFINITE, null);
}
~~~

Reply:
This is a great suggestion. I've made a note to try this out after we merge this PR, since it's pretty orthogonal to the PR, and it will require manual testing since we don't have regression tests for Eclipse usability features like this. The above code is new only because it was converted from xtend to Java.